### DISCUSSION TITLE
Missing functionality

### DISCUSSION THREAD

**Discussion by cloudcompute** [fraktalio/fmodel-java#2] [General]
Hi @idugalic 

As indicated in the Readme file: `This project is in experimental phase`, does this mean it is not production ready, what is missing in it?

I was comparing this code with that of Kotlin, `InternalDecider` is missing.

Regards

**Comment by idugalic** [comment]
Hi @cloudcompute .

Yes, it is still in the experimental phase and has not been published to Maven Central.

`InternalDecider` is included, just with name `_Decider`. 

## What is missing?

- Better naming conventions: `_Decider` -> `InternalDecider`
- Writing tests for the `application` layer components.
- A demo application, utilizing virtual threads in JVM
- A better documentation (JavaDoc and Ref guide)

IMHO, the project is very useful at this point, but I don't consider it production-ready yet. 
I hope to push it there till the beginning of summer. 

PRs are always welcome. 

Best,
Ivan

**Comment by cloudcompute** [comment]
Is it possible to use the fmodel (Kotlin's implementation) from Java code, i.e. defining domain and implementing commands in Java and finally run it on GraalVM?
